src/test/java/com/gui/AbstractTest.java

```diff
@@ -32,11 +47,16 @@ public abstract class AbstractTest {
     @BeforeMethod(alwaysRun = true)
     @Parameters(value = "browser")
     public void setUpTest(String browser) {
-        String hubURL = "http://localhost:4444/wd/hub";
-        DriverFactory driverFactory = new DriverFactory();
-        WebDriver driver = driverFactory.createDriver(browser, hubURL);
-        driver.get("https://automationteststore.com/");
-        threadLocalDriver.set(driver);
+        try {
+            loadProps();
```

Comment:
Every time setUpTest() runs, it reads from the file system. This is inefficient and makes testing harder.